### ISSUE TITLE
Show the actual binary name in the usage help

### ISSUE THREAD

**Issue by pablochacin** [grafana/k6#2850]
### Feature Description

When running a custom k6 binary  (for example, the binaries that are distributed with the xk6-disruptor and xk6-browser extensions) the usage help output makes reference to `k6` command and not to the actual binary name. 

For example, when running `xk6-disruptor` the following message is shown
```
$ xk6-disruptor

          /\      |‾‾| /‾‾/   /‾‾/   
     /\  /  \     |  |/  /   /  /    
    /  \/    \    |     (   /   ‾‾\  
   /          \   |  |\  \ |  (‾)  | 
  / __________ \  |__| \__\ \_____/ .io

Usage:
  k6 [command]

... some output omitted for brevity


Use "k6 [command] --help" for more information about a command
```
This may cause confusion to the user. 

Note: the K6.io banner is correct, because it makes reference to the company brand, not to the name of the command executed by the user.

### Suggested Solution (optional)

The usage help message(s) should show the name of the binary that the user invoked.

**Comment by ariasmn** [comment]
I've had a look at this, and a simple fix would be to create a function in the `consts.go` file to retrieve the binary name. Something like:

``` Go
func BinaryName() string {
	binary, err := os.Executable()
	if err != nil {
		binary = "k6"
	}

	return filepath.Base(binary)
}
```

Seems like related tests run without any issue, however, I don't know if this a good approach. Maybe doing something similar at build time is a better approach?

Related to this, the `version` command displays the k6's version it was built on, but not the extension version. I don't know if you keep track of each extension's version, but maybe is a good idea to show the version for both k6 and the extension.

I would love to work on this, but I think the decision on how to do it should be made by people who know the k6 codebase and the whole ecosystem better than I do.

Let me know what you think!

**Comment by na--** [comment]
This probably shouldn't go to `consts`, since it's not really a constant, if I copy the same k6 binary with a new name (or maybe even if I use a symlink), it will change.

Moreover, we have been trying to get rid of (and avoid adding new) direct dependencies on the `os` package in k6 sub-packages. Instead, we have a [`GlobalState` object](https://github.com/grafana/k6/blob/1d99b0be9194bf9aaf86c070dd0d34604f266abe/cmd/state/state.go#L21-L54) that is populated with the real OS process-external state and, ideally, is the only place where we directly "touch" the `os` package. We also have the [`GlobalTestState` object that wraps `GlobalState`](https://github.com/grafana/k6/blob/1d99b0be9194bf9aaf86c070dd0d34604f266abe/cmd/tests/test_state.go#L23-L34) and is [populated with mocks of the external state](https://github.com/grafana/k6/blob/1d99b0be9194bf9aaf86c070dd0d34604f266abe/cmd/tests/test_state.go#L36-L114), which allows us to easily write [reliable e2e/integration tests](https://github.com/grafana/k6/tree/master/cmd/tests) that can safely run in parallel, without polluting any global state or scopes. 

So, if we implement this issue, `os.Executable()` (or its result) should be exposed in `GlobalState`, which will allow us to have a test that verifies k6 behaves correctly with a different executable path.

**Comment by ariasmn** [comment]
Yeah, you are totally right about the `consts`, this is what a lack of sleep looks like. :facepalm:

Anyways, I tested it again (manually) after implementing the whole logic using the `GlobalState` and:
- Soft link: The source file of the link name is the binary name. If the source file is deleted, the symlink is broken so it won't run (expected behavior I guess).
- Hard link: The hard link name is the binary name (again, as expected).

I guess that the real testing should be done in other runtimes like Windows?

Maybe I'm missing something super obvious, so please forgive my newbie questions, please :disappointed: :

**Comment by na--** [comment]
I think we can count on `os.Executable()` to be sufficiently well tested on the Go side :sweat_smile: Or at least I hope so, since I don't want to maintain a bunch of OS-specific tests about it in our own repo :crossed_fingers: :sweat_smile: 

The good thing is that once we have it mock-able as part of the `GlobalState`, we can add a end-to-end test that k6 behaves correctly with a mocked value for it. For example. a few tests that check if `k6` (without any arguments) and commands like `k6 --help`, `k6 run --help` behave as expected and show whatever we have substituted in our `GlobalTestState`. Most of the current e2e tests are in the [`cmd/tests` folder](https://github.com/grafana/k6/tree/master/cmd/tests), and for example this is a simple one that checks the output of `k6 version`: https://github.com/grafana/k6/blob/1268fad7bde6f0cba8f7159e7bd788331a9f432f/cmd/tests/cmd_run_test.go#L32-L48

_(Though it should probably be moved out of `cmd_run_test`, since it doesn't actually check `k6 run` :sweat_smile: )_

**Comment by ariasmn** [comment]
Ok, perfect! Thank you for the pointers.

Talking about the behaviour of `os.Executable()`, I just read about the possible strange behavior when using symlinks [directly from the source code](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/os/executable.go;l=9). As it states, it's depending on the operating system, so that's why I probably couldn't find a strange behavior in Linux.

Perhaps the best option is to just take the safest path (use `filepath.EvalSymlinks()`) to avoid problems, since the final output in Linux is exactly the same.

**Comment by na--** [comment]
To be honest, I am not sure what the behavior with symlinks should be. Given that official extensions like xk6-browser and xk6-disruptor are meant to eventually be merged back into the k6 core, it's arguable if we should even spend the time to build their own custom semi-official releases with custom binary names... :thinking: 

For me, this issue issue will mostly be helpful for heavy [xk6](https://github.com/grafana/xk6) users that potentially have multiple builds of k6 laying around. 

I am not sure how we should treat symlinks, but unless there are tangible UX or security benefits in trying to follow them, I'd prefer to keep things simple and just use the results of `os.Executable()` :thinking:

**Comment by javaducky** [comment]
Along with the executable name, it would be nice to receive command line args as well. This info could then be passed along to extensions which could then possibly take advantage of them, e.g. script name.

**Comment by na--** [comment]
It's not as simple as that, unfortunately. When you run an archive file, the command line might not tell you what you want to know, which might be the original script path/name. And this should definitely be a separate issue, with a use case that explains how it can be useful.

**Comment by ariasmn** [comment]
Sorry I didn't work on this during a long time, was a bit busy.
Just opened a PR for this issue. (#2916)
